spt.c

```diff
@@ -91,13 +91,12 @@ display_state(int remaining, int suspend)
 		 (suspend) ? "◼" : "▶");
 
 	notify_send(buf);
-	display = 0;
 }
 
 void
 toggle_display(int sigint)
 {
-	display = 1;
+	// used to interrupt nanosleep()
```

Comment:
I think it would be better if we use **C style comments**. As well, if `toggle_display` doesn't do anything, we can just **remove it**, just keep something for the interrupt.

Otherwise, the rest looks fine.

Reply:
@polyzen Nice, thanks a lot to keep track of the man pages as well.

@vaygr Cool that you have replied so fast but there's room for improvement for the patches. :+1:

Reply:
@pickfire appreciated, I corrected the comment part.

regarding non-used `toggle_display` -- that was the first thing I did trying to figure out how to interrupt `nanosleep()` without any handled attached to `SIGUSR1`, and ended up with an empty function. Do you have any suggestions?

Reply:
Hello,

@vaygr I don't think there is another way to define an empty sa_handler, besides an empty function.

Not defining it or using SIG_DFL would let the process handle the default action, for SIGUSR1 it terminates process. SIG_IGN will just ignore SIGUSR1.

Playing around with your PR, I get some funny results. I'm using this config.h for debugging:

```c
static Timers timers[] = {
        /* timer(s)  comment */
        {       10,  "Time to start working!"},
        {       20,  "Time to start resting!"},
};
```

I got this:

```shell
$ spt -n echo
spt Time to start working!
spt Remaining: 00:10 ▶
spt Time to start resting!
spt Remaining: 00:20 ▶
spt Time to start working!
spt Remaining: 00:10 ▶
spt Remaining: 00:20 ▶ # this line should be one row below
spt Time to start resting!
spt Remaining: 00:10 ▶  # this line should be one row below
spt Time to start working!
spt Time to start resting!
spt Remaining: 00:20 ▶
spt Time to start working!
```

FWIW, this may happen because of notify_send()/spawn() using fork()/execlp() which might not run in the same order. But this may not be is a real issue though.

Anyway, IMHO I'm not really sure if this is a good idea to automatically display the timer remainder & state on toggling suspend/continue. It may not be a need for everyone.

Wouldn't it be better to use a wrapper successively sending both signals USR1/USR2 for this need?

Reply:
@pips- yeah, I also checked `SIG_IGN` and `SIG_DFL` cases yesterday.

With libnotify enabled I couldn't reproduce what you got, btw.

I just think even we do not want to display *the real state*, it is still probably a good idea to display some message once you trigger it, e.g. "Paused" and "Continued".

Reply:
Yeah maybe, I would not rely on this, but that's a matter of taster.

However, what about using some `prevsuspend` variable and display Pause/Continued when it differs from `suspend` in the main loop,  updating it with the new state? This would be a bit more explicit on it purpose, no?

Reply:
OTOH if one desires the remaining time/state notification on state toggle, using a wrapper would then result in two notifications on toggle.

Reply:
@pips- Thanks a lot for suggesting an idea to @vaygr. I don't even know how to do that placebo function.

@vaygr About taste, I prefer to keep the play/pause separate from displaying the state. Of course I keep track of the state when I sent that signal (if not why would I send a signal in the first place).

@polyzen @pips- True though that sending two signal here is the best case though, @vaygr what do you think? As for me, I might see this as an deprovement rather than an improvement.

Reply:
@polyzen I'm not sure to understand what you mean by using a wrapper, sending both signals, would result in two notifications on toggle? I mean, currently using @pickfire's HEAD, the only signal triggering a notification is SIGUSR1 for timer state display, not SIGUSR2.

@pickfire @vaygr 
In order to bring together the better of the two worlds :) maybe we can use something like this:

https://gist.github.com/pips-/6588a5ada1990b7cd39c12411d481163

(Apply to @pickfire's master/HEAD)

It keep the SIGUSR1 display timer state logic aside from SIGUSR2 toggling state logic, but add a notification on effective toggling to display state changes.

What do you think?

Reply:
@pips, to clarify: I personally would also like to see the time remaining when toggling. The only way to do so, without patching, would be with a wrapper, and with the changes considered, this would result in two notifications.

Reply:
@polyzen okay, thanks for clarification :) Indeed, this would push two notifications with the considered PR. When I was referring to make a wrapper, I thought of a wrapper without this PR.

But with my previous gist, we may also change the `notify_send("Paused ◼");`/`notify_send("Continued ▶");` by `display_state()`, if preferable.

Reply:
2 things:

1. 2 consecutive signals with leaving the code as is is not a solution, but rather a workaround (which I currently use too) because it still doesn't fix the original behavior: no reaction on the signal.
2. @pips- that's right, and it will be more clear to have 1 notification instead of 2. Should we need to display remaining time, we will send the second signal if we do not find this information _that_ important.

Reply:
So this leaves the question @polyzen raised open: do we find remaining time display _that_ important or we should keep toggle/state-display logic strictly separate and leave those who wanted it with a wrapper option about sending 2 signals?

Reply:
Is it feasible to make this a configuration option?

Reply:
Eg.:
```
-t notifytoggle
Show time and/or state on toggle: none(default), state, both(time and state)
```

Reply:
Hm, maybe I wasn't clear enough, sorry :/

I do not really think displaying Paused/Continued OR `display_state()` on execution toggling is annoying. I really can get over it or adding an command line option may be is a solution. :) 

But, I'm just advocating we do not rely on an empty function to exit `nanosleep()` when receiving SIGUSR1 thus forcing `display_state()` each time in the main loop. 

This is what I mean by keeping logic separated:

* SIGUSR1: `toggle_display()` to call `display_state()` only ;
* SIGUSR2 : toggle execution and explicitly does `display_state()` in code (or Pause/Continued, whatever).

Not something like:

* SIGUSR1: wake up `nanosleep()` by doing nothing in order to `display_state()` at the beginning of the main loop ;
* SIGUSR2: toggle execution and it turns out we `display_state()` at the beginning of the main loop.

Do you know what I mean?

Reply:
@pips- yeah, I think we were just focusing on 2 different things: you -- keeping logic clear; me, @polyzen -- mostly just _ability_ to display state on toggle, which I think what @pickfire wouldn't prefer.